addOns/exim/src/main/java/org/zaproxy/addon/exim/log/LogsImporter.java

```diff
@@ -173,12 +173,18 @@ private boolean processInput(File newFile, LogType logChoice) {
         return true;
     }
 
-    private static List<String> readFile(File file) throws IOException {
+    /**
+     * This method is {@code protected} to facilitate Unit Testing. It is not expected to be used
+     * directly.
+     *
+     * @throws IOException
+     */
+    protected static List<String> readFile(File file) throws IOException {
         List<String> parsed = new ArrayList<>();
         Charset charset = StandardCharsets.US_ASCII;
         BufferedReader reader = Files.newBufferedReader(file.toPath(), charset);
         Scanner sc = new Scanner(reader);
-        sc.useDelimiter(Pattern.compile("====\\s[0-9]*\\s=========="));
+        sc.useDelimiter(Pattern.compile("====?\\s?[0-9]*\\s=========="));
```

Comment:
I did identify one issue with this implementation (though it always would have existed). There's no way for the scanner to know how many things it should be finding, so if it doesn't find a match it treats the entire content as a single 'blob'. (I guess similarly if there were multiple good delimiters with a bad one in the middle then the 'bad' message will end up "stuck" together with the one ahead of it.)

This shouldn't be a major issue as it should only come up with a user manually creates the file/entries themselves.

Reply:
There's also the (unlikely) case of the delimiters being in the content of the messages. We should look at deprecating this format.